packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml

```
@@ -8,102 +8,177 @@ processors:
     if: ctx._temp_?.unparsed?.message != null && ctx._temp_?.unparsed?.message != ""
     tag: grok_unparsed_message
     patterns:
-#   # SRX System log patterns
-    - '^(?:%{PROG:syslog_program}|-)?\s(?:%{POSINT:syslog_pid}|-)?\s(?:%{WORD:tag}|-)?\s?((\[([^=]+?\s)?%{DATA:_temp_.system_structured}\"\])|-)?\s?(%{CUSTOM_PROG} )?%{CUSTOM_SYSTEM_TAG:_temp_.sub_tag}:%{DATA:_temp_.unparsed.syslog_structured}\s?$'
-    - '^(?:%{PROG:syslog_program})?\s(?:%{POSINT:syslog_pid})?\s(?:%{WORD:tag})?\s?(\[([^=]+?\s)?%{DATA:_temp_.system_structured}\"\])?\s?%{DATA:message}\s?$'
-    - '^(%{CUSTOM_PROG} )?%{CUSTOM_SYSTEM_TAG:_temp_.sub_tag}:%{DATA:_temp_.unparsed.syslog_structured}\s?$'
+    # 1. SRX System Structured Brief
+    - '^(?:%{PROG:syslog_program}|-)?\s(?:%{POSINT:syslog_pid}|-)?\s(?:%{WORD:tag}|-)?\s([-]+\s)?%{GREEDYDATA:_temp_.unparsed.system_structured_brief}\s?$'
+    # 2. SRX System Unstructured
     - '^%{GREEDYDATA:message}$'
-    pattern_definitions:
-        CUSTOM_DATE: "%{TIMESTAMP_ISO8601}|(%{MONTH}%{SPACE}+%{MONTHDAY}%{SPACE}+%{TIME})"
-        CUSTOM_PROG: "(%{DATA:_temp_.to_be_parsed})?%{PROG:_temp_.syslog_sub_program}(\\[%{POSINT:syslog_pid}\\]|\\s*\\(pid=%{DATA:syslog_pid}\\))?:"
-        CUSTOM_SYSTEM_TAG: "[A-Za-z_]+"
-
-# split k-v fields inside structured data
-- kv:
-    field: _temp_.system_structured
-    if: 'ctx._temp_?.system_structured != null && ctx._temp_?.system_structured != ""'
-    field_split: " (?=[a-z0-9\\_\\-]+=)"
-    value_split: "="
-    prefix: "juniper.srx."
-    ignore_missing: true
-    ignore_failure: false
-    trim_value: "\""
-
-
-# Converts all kebab-case key names to snake_case
-- script:
-    lang: painless
-    if: ctx.juniper?.srx != null
-    source: >-
-        ctx.juniper.srx = ctx?.juniper?.srx.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().replace('-', '_'), e -> e.getValue()));
 
+# Parse different types of system_structured_brief messages
+- grok:
+    if: "ctx._temp_?.unparsed?.system_structured_brief != null && ctx._temp_?.unparsed?.system_structured_brief != ''"
```

Here we could avoid checking if the field exists for the second term
```suggestion
    if: "ctx._temp_?.unparsed?.system_structured_brief != null && ctx._temp_.unparsed.system_structured_brief != ''"
```

---

updated as suggested